main/pipeline/pipeline.cc

```diff
@@ -1222,25 +1222,62 @@ void computeFileHashes(const vector<shared_ptr<core::File>> files, spdlog::logge
     }
 }
 
-bool cacheTreesAndFiles(const core::GlobalState &gs, vector<ast::ParsedFile> &parsedFiles,
+bool cacheTreesAndFiles(const core::GlobalState &gs, WorkerPool &workers, vector<ast::ParsedFile> &parsedFiles,
                         const unique_ptr<OwnedKeyValueStore> &kvstore) {
     if (kvstore == nullptr) {
         return false;
     }
 
     Timer timeit(gs.tracer(), "pipeline::cacheTreesAndFiles");
-    int i = -1;
-    bool written = false;
-    for (auto &pfile : parsedFiles) {
-        i++;
-        if (!pfile.file.exists()) {
-            continue;
+
+    // Compress files in parallel.
+    shared_ptr<ConcurrentBoundedQueue<ast::ParsedFile *>> fileq =
```

Comment:
Why not use `auto` here?

Reply:
Good question. I mindlessly cargo culted from other similar lines of code. Can fix.